### ISSUE TITLE
Trying to use a wildcard (*) in Include / Exclude folder list doesn't work and produces an error

### ISSUE THREAD

**Issue by owbosh** [husjon/obsidian-file-cleaner-redux#79]
I think a fairly understandable use-case would be to include specific folder names throughout your vault, not just at specific paths, like this: 

![image](https://github.com/user-attachments/assets/54622415-f66e-4b50-9918-ee2dd3862e24)

However, this does not work and also produces an error in the console:

![image](https://github.com/user-attachments/assets/19cb67ec-cd35-4a0c-9ddb-07c06426b47a)

Alternatively, instead of supporting the wildcard in the Include / Exclude folder list, another idea would be to provide an option for only checking inside the attachment folder(s) that is configured in the Obsidian settings:

![image](https://github.com/user-attachments/assets/522b9133-1035-43df-8a33-c19f2cf4e5d2)

**Comment by husjon** [comment]
Since the excluded folders lines are part of a regular expressions, you'd have to use `.*`.
This is also why the error says `Nothing to repeat`, the `*` quantifier requires something to repeat (in this case `.`, meaning any character).

Example with description from regex101:
Note the parentheses are not required, just added here for show the grouping of the match
![image](https://github.com/user-attachments/assets/f316769b-8489-432f-a889-22149c10d1b9)


I don't remember exactly why I decided on it being regular expressions so I might tag this as a bug.

**Comment by owbosh** [comment]
Ah, it does work, thank you! Please consider adding a little text to the description of the Include / Exclude folders setting box that explains this, it's very useful.

**Comment by husjon** [comment]
I agree, I don't remember exactly why I went for regular expressions but it is way more powerful than regular globbing (using `*` as wildcards).
I'll leave the issue open as a reminder to add explanation to the description.

**Comment by husjon** [comment]
I just had a look at the code for the folder exclusion and I think I now know why I chose the regular expression, it was just as simple as me completely forgetting about the `.startsWith` function for strings.
This is literally what the current exclusion check does now, checks if the string starts with the path you've specified in the settings (although with the added benefit of allowing regular expressions).

`.startsWith` cannot do wildcards, so I guess it just has to stay as regex for the foreseeable future :)

I will update the description to inform about this to make it a bit more clear.

**Comment by husjon** [comment]
I've prepared a pull request for this issue expanding the description a bit to inform about regular expression support, see #84
This will be part of version 1.4.1 which will be released shortly.
This issue will be closed once done.
